darwin/DarwinMachine.h

```diff
@@ -19,6 +19,9 @@ typedef struct DarwinMachine_ {
 
    host_basic_info_data_t host_info;
    vm_statistics_data_t vm_stats;
+#if defined(__arm64__)
+   vm_statistics64_data_t vm_stats64;
```

Comment:
This structure does not seem to be exclusive to ARM64. Would it be used in Intel Macs, too? A reference to Apple documentation would be useful.

Reply:
You are right. According to the documentation:
https://developer.apple.com/documentation/kernel/vm_statistics64_data_t
https://developer.apple.com/documentation/kernel/vm_statistics_data_t

**vm_statistics64_data_t** can be used after macOS 10.6+
**vm_statistics_data_t** can be used after macOS 10.0+

Reply:
@SuCicada  I prefer keeping a fallback mechanism when `vm_statistics64_data_t` is not available. There is no reason to remove 32-bit Mac support (htop documentation doesn't state a minimum required version of macOS).

Reply:
Haven't looked at the details of the implementation yet, but would it be feasible to check for `vm_statistics64_data_t` and `vm_statistics_data_t` in `configure` and use the 64 bit version everywhere when available? And if none of both is available, we either fall back entirely or bail out at compile time.

Reply:
@BenBE Since it is said that `vm_statistics_data_t` is available in OS X 10.0 (the very first Mac OS X), these is no need to check this one in particular in `configure`. Just checking for `vm_statistics64_data_t` is enough.

Reply:
So I will restore support for vm state 32.

Reply:
Exactly: Use `vm_statistics64_data_t` when available, otherwise fall back to `vm_statistics_data_t`.